template/src/routes/profile/index.tsx

```diff
@@ -1,54 +1,46 @@
-import { Component, h } from "preact";
+import { h } from "preact";
+import { useCallback, useEffect, useState } from "preact/hooks";
 import * as style from "./style.css";
 
 interface Props {
     user: string;
 }
 
-interface State {
-    time: number;
-    count: number;
-}
-export default class Profile extends Component<Props, State> {
-    public state = {
-        time: Date.now(),
-        count: 10
-    };
+const Profile: preact.FunctionalComponent<Props> = props => {
+    const { user } = props;
+    const [time, setTime] = useState<number>(Date.now());
+    const [count, setCount] = useState<number>(0);
 
-    public timer?: number;
+    let timer: number;
 
     // gets called when this route is navigated to
-    public componentDidMount() {
-        // start a timer for the clock:
-        this.timer = window.setInterval(this.updateTime, 1000);
-    }
+    useEffect(() => {
+        timer = window.setInterval(() => setTime(Date.now()), 1000);
 
-    // gets called just before navigating away from the route
-    public componentWillUnmount() {
-        clearInterval(this.timer);
-    }
+        // gets called just before navigating away from the route
+        return () => {
+            clearInterval(timer);
+        };
+    }, []);
 
     // update the current time
-    public updateTime = () => {
-        this.setState({ time: Date.now() });
-    };
-
-    public increment = () => {
-        this.setState({ count: this.state.count + 1 });
-    };
-    public render({ user }: Props, { time, count }: State) {
-        return (
-            <div class={style.profile}>
-                <h1>Profile: {user}</h1>
-                <p>This is the user profile for a user named {user}.</p>
-
-                <div>Current time: {new Date(time).toLocaleString()}</div>
-
-                <p>
-                    <button onClick={this.increment}>Click Me</button> Clicked{" "}
-                    {count} times.
-                </p>
-            </div>
-        );
-    }
-}
+    const increment = useCallback(() => {
```

Comment:
also I think as long as we don't pass this to another component it's better to just have a bare arrow function (not wrapped in `useCallback`)

For reference https://kentcdodds.com/blog/usememo-and-usecallback